pilot/pkg/networking/core/v1alpha3/cluster_test.go

```diff
@@ -788,13 +791,15 @@ func TestConditionallyConvertToIstioMtls(t *testing.T) {
 		Sni:               "custom.foo.com",
 	}
 	tests := []struct {
-		name            string
-		tls             *networking.TLSSettings
-		sans            []string
-		sni             string
-		proxy           *model.Proxy
-		serviceMTLSMode authn_v1alpha1_applier.MutualTLSMode
-		want            *networking.TLSSettings
+		name                      string
+		tls                       *networking.TLSSettings
+		sans                      []string
+		sni                       string
+		proxy                     *model.Proxy
+		serviceMTLSMode           authn_v1alpha1_applier.MutualTLSMode
+		mtlsReady                 bool
+		want                      *networking.TLSSettings
+		enableTransportSocketMtls bool
```

Comment:
renamed to `mtlsBySocketMatch`? I mean everything is **transport socket** already per se, using the prefix of a Cluster.proto field can be confusing